### DISCUSSION TITLE
The devtool is not working in Next.js 14 App router.

### DISCUSSION THREAD

**Discussion by huraim** [jotaijs/jotai-devtools#125] [Q&A]
Hello,
I'm trying to install devtools in the Next.js 14 app router, but although the UI is displayed properly, when I try to open the devtools, I encounter the following error. What could be the issue?

```
Uh-oh, something went wrong.
If you believe this to be a bug, please file an issue on [Jotai DevTool's GitHub repo](https://github.com/jotaijs/jotai-devtools/issues) with a minimal reproduction and the following error

TypeError: values.map is not a function
    at ErrorFallback (webpack-internal:///(app-pages-browser)/./node_modules/jotai-devtools/dist/chunk-O6NZVQU4.esm.mjs:8273:204)
    at renderWithHooksAgain (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:9869:28)
    at renderWithHooks (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:9756:32)
    at updateFunctionComponent (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:14159:32)
    at beginWork$1 (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom/cjs/react-dom.development.js:15948:32)
```

```ts
/* next,confing.mjs */

const nextConfig = {
  transpilePackages: ["jotai-devtools"],
  experimental: {
    swcPlugins: [
      ["@swc-jotai/debug-label", {}],
      ["@swc-jotai/react-refresh", {}],
    ],
  },
};
```

```tsx
/* layout.tsx */
import JotaiDevtool from "../_components/jotai-devtool";
...
return (
<html lang="en">
  <body>
    <Provider>
      <JotaiDevtool />
      {children}
    </Provider>
  </body>
</html>
)
```

```tsx
/* jotai-devtool.tsx */
"use client";

import { DevTools } from "jotai-devtools";

export default function JotaiDevtool() {
  return <DevTools theme="dark" />;
}
```

**Comment by arjunvegda** [comment]
That's strange! I can take a look at it. Could you please create a repro on CodeSandbox?

**Reply by huraim** [reply]
Please take a look and let me know if I've done something wrong.
Here is the address for CodeSandbox.
[CodeSandbox-NextJS14.1-AppRouter-JotaiDevtool](https://codesandbox.io/p/devbox/jotai-devtool-in-app-router-h278vx?file=%2Fapp%2Fjotai-provider.tsx%3A17%2C16&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt0pk58g0007356hcwx3aaum%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt0pk58g0002356hfcdj3ki0%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt0pk58g0004356h45myzttm%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt0pk58g0006356h927lqrui%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B58.52156254190854%252C41.47843745809146%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt0pk58g0002356hfcdj3ki0%2522%253A%257B%2522id%2522%253A%2522clt0pk58g0002356hfcdj3ki0%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt0pzc1b0002356hwcgx4bg8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A24%252C%2522startColumn%2522%253A21%252C%2522endLineNumber%2522%253A24%252C%2522endColumn%2522%253A21%257D%255D%252C%2522filepath%2522%253A%2522%252Fapp%252Flayout.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clt0qr2yh0002356he4ih6r81%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A17%252C%2522startColumn%2522%253A16%252C%2522endLineNumber%2522%253A17%252C%2522endColumn%2522%253A16%257D%255D%252C%2522filepath%2522%253A%2522%252Fapp%252Fjotai-provider.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clt0qr2yh0002356he4ih6r81%2522%257D%252C%2522clt0pk58g0006356h927lqrui%2522%253A%257B%2522id%2522%253A%2522clt0pk58g0006356h927lqrui%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt0pk58g0005356h8zpe1jf0%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clt0pk58g0005356h8zpe1jf0%2522%257D%252C%2522clt0pk58g0004356h45myzttm%2522%253A%257B%2522id%2522%253A%2522clt0pk58g0004356h45myzttm%2522%252C%2522activeTabId%2522%253A%2522clt0pk58g0003356hvlq3qvp4%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt0pk58g0003356hvlq3qvp4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clt0pm2xg008a356hxofnbl7z%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clt0pm4lx012cdige022707oq%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by arjunvegda** [reply]
Thank you! CSB is breaking for me, its not able to find `useAtomValue` from `jotai/react` which is strange. I'll take a deep look later.

**Reply by mnpqraven** [reply]
I think nextJS 14.1 is also breaking other stuffs for jotai. `atomWithStorage` also doesn't seem to work properly, I'll try to reproduce it and if successful will cut an issue on the jotai repo

**Reply by mnpqraven** [reply]
I have created an issue and provided the reproduction project in #127

**Reply by arjunvegda** [reply]
Thanks @mnpqraven  — @huraim, let's continue our discussion there https://github.com/jotaijs/jotai-devtools/issues/127#issuecomment-1964844049

**Comment by mnpqraven** [comment]
I have the same problem with ^14.1 while my other project running 14.0.2 still works so I'd recommend downgrading to 14.0 for now

**Reply by huraim** [reply]
Oh... Looks like there might be a problem with version 14.1... Thank you for your response. I'll keep it in mind
